### PR TITLE
Multi-threaded API v2 handler

### LINKED ISSUE #4963: Implement multi-threading in the API v2 handler
For reference, the current WebHandler:
https://github.com/pymedusa/Medusa/blob/1b62217a9ea7dbceae924d8f3dd2bcb46caaecda/medusa/server/web/core/base.py#L211-L262

### PR DESCRIPTION
**Pull request by sharkykh**
Fixes #4963

### CODE REVIEW

File: medusa/server/api/v2/base.py [hunk 1]
```diff
@@ -167,12 +234,13 @@ def create_app_handler(cls, base):
 
     def _handle_request_exception(self, e):
```
Comment (p0psicles) [inline medusa/server/api/v2/base.py]: can `e` be changed to `error`?

**Comment by p0psicles** [comment]
Let's keep it up2date with develop?

**Comment by sharkykh** [comment]
There's still an issue that the errors generated inside the routes are not being handled correctly.
To test this just raise an exception inside one of the routes, `GET /config` for example..

**Comment by sharkykh** [comment]
https://github.com/pymedusa/Medusa/blob/688f4a67a00f47253524b54b2a417b5c88122ef9/medusa/server/api/v2/base.py#L89-L95

I still think that solution is a bit crude, but I don't have any better ideas at the moment.
That said, I do believe it's stable now, so I guess you can go ahead and merge...

**Comment by Rouzax** [comment]
@sharkykh 
This also fulfills https://github.com/pymedusa/Medusa/issues/3351
Does this mean that when I call the API it will give an acknowledge directly or will it still wait for the process to complete and it can just handle more "threads" at a time?

**Comment by p0psicles** [comment]
It can handle more threads. It will always wait for the process thats providing its result

**Comment by p0psicles** [comment]
Oh this is unrelated to 3351

**Comment by Rouzax** [comment]
Is the GUI still unresponsive when post processing? 
If it is responsive that actually fulfils my request in the original #3351

**Comment by Rouzax** [comment]
I do it when a download is done by calling the API. So in occasion when I browse Medusa it will hang because it is post processing in the back end.
Not a big issue. Not sure with the new threaded post processing if that is still the case.

**Comment by labrys** [comment]
One thing to watch for is locked threads due to failed post processing.  They will lock up resources.  This happens most commonly when trying to postprocess an invalid rar file.

**Comment by Rouzax** [comment]
Medusa will never encounter a rare in my environment as I already unrar files with a post process script that is kicked off when qbitorrent finishes the download, the last step is calling the Medusa api to post process.

**Pull request merged**